in_toto/runlib.py

```diff
@@ -588,7 +593,33 @@ def in_toto_run(
 
     if use_dsse:
         LOG.info("Generating link metadata using DSSE...")
-        link_metadata = Envelope.from_signable(link)
+
+        if use_statement == False:
```

Comment:
Potential footgun: if I set `use_statement` (or `use_attestation`) to True, I shouldn't be expected to set `use_dsse` to True as well. I see the CLI handles this but this is an issue with direct API use. I suggest we move `use_attestation` out of the `use_dsse` block and only support DSSE there. The `use_dsse` conditional block should only be for legacy links, and something we phase out in a future version as we deprecate legacy links in favour of link predicates.